pyhf/commandline.py

```diff
@@ -83,7 +83,9 @@ def json2xml(workspace, xmlfile, specroot, dataroot):
 )
 @click.option('--measurement', default=None)
 @click.option('-p', '--patch', multiple=True)
-def cls(workspace, output_file, measurement, patch):
+@click.option('--optimizer')
+@click.option('-n', '--max-iterations', default=1000)
+def cls(workspace, output_file, measurement, patch, optimizer, max_iterations):
```

Comment:
going to be tricky since different optimizers will have different opts. One appraoch I used in https://github.com/yadage/yadage is to have a mainswitch to select type and then one for setting optiosn

```
... --backend xyc --backendopt hello=world --backendopt foo=bar --backendopt baz=qux
```

Reply:
This is now done with the `--optconf` option.